sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -109,19 +109,23 @@ class EthService(val scNodeViewHolderRef: ActorRef, val nvtimeout: FiniteDuratio
     }
   }
 
-  private def doCall[A](nodeView: NV, params: TransactionArgs, tag: String)(fun: (Array[Byte], AccountStateView) ⇒ A): A = {
+  private def doCall[A](nodeView: NV, params: TransactionArgs, tag: String, forwardErrors: Boolean)(fun: (Array[Byte], AccountStateView) ⇒ A): A = {
     getStateViewAtTag(nodeView, tag) { tagStateView =>
-      try {
-        val msg = params.toMessage(tagStateView.getBaseFee)
-        fun(tagStateView.applyMessage(msg, new GasPool(msg.getGasLimit)), tagStateView)
-      } catch {
-        // throw on execution errors, also include evm revert reason if possible
-        case reverted: ExecutionRevertedException => throw new RpcException(new RpcError(
-          RpcCode.ExecutionError.getCode, reverted.getMessage, Numeric.toHexString(reverted.revertReason)))
-        case err: ExecutionFailedException => throw new RpcException(new RpcError(
-          RpcCode.ExecutionError.getCode, err.getMessage, null))
-        case err: TransactionSemanticValidityException => throw new RpcException(new RpcError(
-          RpcCode.ExecutionError.getCode, err.getMessage, null))
+      val msg = params.toMessage(tagStateView.getBaseFee)
+      if (forwardErrors) {
+        fun(tagStateView.applyMessage(msg, new GasPool(msg.getGasLimit), true), tagStateView)
```

Comment:
With `forwardErrors == true` this whole function doesn't really do anything anmore :grinning: 
Lets move the exception mapping to `applyOnAccountStateView`. That way it would be wrapped around every relevant call on the outside. [See line 60](https://github.com/HorizenOfficial/Sidechains-SDK/pull/526/files#diff-9f3a238171c2b2e81d34f037a29411b0fd262d527fcbb24883ed13db1676aedbR60)